docs/source/traitsui_user_manual/handler.rst

```diff
@@ -45,21 +45,121 @@ means of the UIInfo object.
+on editors via the **sync_to_view**, **sync_from_view** and **sync_with_view**
+trait metadata.  Note that not every trait on every editor can react to changes:
+some values are only used at editor creation time; however all editors support
+dynamically changing the **enabled**, **visible** and **invalid** traits.  This
+feature can sometimes allow developers to avoid having to create a custom Handler
+subclass.
+
+See the :github-demo:`Invalid state handling <Advanced/Invalid_state_handling.py>`
+example which demonstrates how to use this mechanism to control the invalid state
+of a dialog based on the value of multiple editors.
+
+.. _backstage-the-ui-object:
+
+Backstage: The UI Object
+------------------------
+
+As opposed to the very dynamic UIInfo object, the UI class provides an object
+which ties together the various objects that are involved in a TraitsUI GUI:
+the View, the context, the Handler, the underlying toolkit controls and the
+shared state of the GUI.  It also has some life-cycle and useful utility methods
+that can be useful when working from a handler.
+
+The UI object is returned as the result of a call to edit_traits(), and as noted
+in the previous section, is available as the **ui** attribute of the UIInfo object
+that is passed to most handler methods.
+
+.. rubric:: Attributes of UI, by category
+
+TraitsUI core
+   view:
+      View template used to construct the user interface.
+   handler:
+      Handler object used for event handling.
+   context:
+      Dictionary of objects that the UI is editing.
+   info:
+      UIInfo object containing context or editor objects
+   parent:
+      The parent UI (if any) of this UI.
+
+Toolkit
+   control:
+      Panel or dialog associated with the user interface.
+   owner:
+      Toolkit-specific object that "owns" **control**
+
+GUI state
+   id:
+      The unique ID for this UI for persistence.
+   title:
+      Title of the dialog, if any.
+   icon:
+      The ImageResource of the dialog icon, if any.
+   key_bindings:
+      The KeyBindings object (if any) for this UI.
+   result:
+      Result from a modal or wizard dialog.
+   modified:
+      Have any modifications been made to UI contents?
+   updated:
+      Event when the user interface has changed.
+   history:
+      Undo and Redo history.
+   errors:
+      The number of currently pending editor error conditions.
+      Note that changing this must be done very carefully to avoid
+      permanent error states.
+   destroyed:
+      Set to True when the UI has finished being destroyed.
+
+
+.. rubric:: Useful UI methods
+
++---------------------------+--------------------------------------------------+
+|Method                     |Purpose                                           |
++===========================+==================================================+
+|dispose(result, abort)     |Disposes of the UI.  This can be called to close a|
```

Comment:
nit: can we have spaces around the content in the table entries (as in the examples [here](https://docutils.sourceforge.io/docs/user/rst/quickref.html#tables))? I don't think it makes a difference to the rendered result, but it makes it more readable in the source.

Reply:
I was following the style of other tables.  Can fix all.

Reply:
Up to you; nitpick level, not important.

Reply:
I think I am not going to do it in this PR.  There's probably a more thorough overhaul of documentation needed, but not today.